File: CV_tasks/Cpp/display_video.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

int main() {

	cv::VideoCapture webcam(0);

	if (!webcam.isOpened()) {
		std::cerr << "Error: Couldn't take input from webcam." << std::endl;
		return 1;
	}

	cv::namedWindow("webcam", cv::WINDOW_NORMAL);

	while(true) {
		cv::Mat frame;
		webcam >> frame;

		if (frame.empty()) {
			std::cerr << "Error: Couldn't read a frame." << std::endl;
			break;
		}

		cv::imshow("webcam", frame);

		if (cv::waitKey(1) == ' ') { // space
			break;
		}
	}

	webcam.release();
	cv::destroyAllWindows;

    return 0;
}

File: CV_tasks/py/display_video.py
import cv2

def main():
    webcam = cv2.VideoCapture(0)
    if not webcam.isOpened():
        print("Error: couldn't open webcam.")
        exit()

    while True:
        valid_frame, frame = webcam.read()
        if not valid_frame:
            break

        cv2.imshow("WebCam", frame)
        if cv2.waitKey(1) == ord(' '):
            break

    webcam.release()
    cv2.destroyAllWindows()

if __name__ == "__main__":
    main()